### DISCUSSION TITLE
Create new Callback Types

### DISCUSSION THREAD

**Discussion by GuilhermePessoni** [apache/airflow#39807] [Ideas]
### Description

In versions 2.7.3 or lower, in the process of sending callbacks, they are sent in each taskgroup, and not by dag's.
The proposal is to create a new callback flow where sending takes place by each dag and not by each Taskgroup

### Use case/motivation

In these cases, Airflow sends 8 callbacks, 1 per taskgroup, the purpose is send only 1 callback alert
![image](https://github.com/apache/airflow/assets/130606573/f5666af4-79c7-44a8-ada4-31ce0b91e984)

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by Taragolis** [comment]
TaskGroup it is a UI / Namespace feature, and do not have own states, like a DAG or TaskInstance have, that mean without [significant changes](https://cwiki.apache.org/confluence/plugins/servlet/mobile#content/view/89066602) in Airflow codebase it is unlikely could be implemented.

I would recommend to start discussion in Dev List first: https://airflow.apache.org/community/
